synapse/lib/cell.py

```diff
@@ -1876,6 +1890,10 @@ async def _addDriveItem(self, info, path=None, reldir=s_drive.rootdir):
         if self.drive.hasItemInfo(iden): # pragma: no cover
             return await self.drive.getItemPath(iden)
 
+        perm = info.pop('perm', None)
+        if perm:
+            info['permissions'] = perm
```

Comment:
does this also need to `setdefault` for users/roles?

Reply:
The additemInfo call right after this  does something like that?

https://github.com/vertexproject/synapse/pull/4366/files#diff-92a0aee96034cd4a64ee25e72012a4ea071dd2e6fe6b07113f32583f29eb414aR292

Reply:
Ah, gotcha

Reply:
well that sets it if permission isn't there at all, but wouldn't do it if the permission key was there but didn't have users/roles?  (i'm not sure how that would happen, was mostly just going off the behavior in the migration)

Reply:
Yea, I'm not sure how that would happen either and shouldn't cause a problem, but I can call setdefault.

Reply:
(yea, the previous schema enforce would prevent having a users key without a roles key, but the setdefault isn't bad to have)